test_runner/src/main/kotlin/ftl/run/platform/common/AfterRunTests.kt

```diff
@@ -38,8 +38,8 @@ internal suspend fun afterRunTests(
 }
 
 private fun List<TestMatrix>.toSavedMatrixMap() = this
-    .map { matrix -> matrix.testMatrixId to SavedMatrix(matrix) }
-    .toMap()
+    .map { matrix -> matrix.testMatrixId to createSavedMatrix(matrix) }
+    .toMap().toMutableMap()
```

Comment:
```suggestion
```suggest
    .associate { matrix -> matrix.testMatrixId to createSavedMatrix(matrix) }
    .toMutableMap()
```
```